### ISSUE TITLE
Bug: [lines-around-comment] afterHashbangComment option is not supported on TS

### ISSUE THREAD

**Issue by IronGeek** [eslint-stylistic/eslint-stylistic#75]
Cross-posting issue from typescript-eslint/typescript-eslint#6636

---

### Playground Link
https://typescript-eslint.io/play/#ts=5.0.1-rc&sourceType=module&code=MQQgZg9hBQBuCGAnABAE3gT2QXmQIgHNEBTeAFz2iA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6RplWgQ2gHtYTACa0yfALbjELdGADa4bDkTR+0SABpFSrEuyRuAM3wqAEt2QALAEbcmAcwDCEqTIz44ibdgC+3gLqKfj5AA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code
```ts
#!foo
var day = "great"
```

### ESLint Config
```js
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/lines-around-comment": [
      "error",
      {
        "afterHashbangComment": true
      }
    ]
  },
};
tsconfig
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result
Should not crash, should warn about missing blank line after hashbang comment

### Actual Result
Configuration for rule "@typescript-eslint/lines-around-comment" is invalid:
Value {"afterHashbangComment":true,"beforeBlockComment":true,"afterBlockComment":false,"beforeLineComment":false,"afterLineComment":false,"allowBlockStart":false,"allowBlockEnd":false} should NOT have additional properties.

### Additional Info
The playground example doesn't crash, but CLI crashes.
